src/main/kotlin/com/gamingbacklog/api/gamingbacklogapi/clients/IGDBClient.kt

```diff
@@ -26,21 +19,20 @@ class IGDBClient {
   val authUri = "https://id.twitch.tv/oauth2/token"
   val baseUri = "https://api.igdb.com/v4/"
 
+
   /**
    * Generates a token
    */
   fun authenticate(): Credentials {
-    val request: HttpRequest = HttpRequest.newBuilder()
-      .uri(URI.create("$authUri?client_id=$clientId&client_secret=$clientSecret&grant_type=client_credentials"))
-      .POST(HttpRequest.BodyPublishers.noBody())
-      .build()
-    try {
-      val response = HttpClient.newHttpClient().send(request, HttpResponse.BodyHandlers.ofString())
-      return Gson().fromJson(response.body(), Credentials::class.java)
-    } catch (ex: Exception) {
-      ex.printStackTrace()
+    val response = externalAPIClient.postExternalCall(
+      "$authUri?client_id=$clientId&client_secret=$clientSecret&grant_type=client_credentials"
+    )
+    return if (response.statusCode() == 200) {
+      Gson().fromJson(response.body(), Credentials::class.java)
```

Comment:
There's no Credentials class in Kotlin?

Reply:
This is how Kotlin references classes. Instead of just doing `Credentials.class` the way Java does it, we need this syntax. I believe the `.class` is a decompiled Java thing, so this is the syntax that requires Kotlin to decompile classes (at least that's my understanding)